homeassistant/components/yamaha/media_player.py

```diff
@@ -260,10 +284,15 @@ def build_source_list(self):
 
         self._attr_source_list = sorted(
             self._source_names.get(source, source)
-            for source in self.receiver.inputs()
+            for source in self.zctrl.inputs()
             if source not in self._source_ignore
         )
 
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this media_player."""
+        return self._attr_unique_id or ""
+
```

Comment:
This is the default implementation, please revert

Reply:
will do with the uid fix pull next